clients/githubrepo/client.go

```diff
@@ -52,31 +55,34 @@ func (client *Client) InitRepo(owner, repoName string) error {
 	url = strings.Replace(url, "{/ref}", client.repo.GetDefaultBranch(), 1)
 	req, err := http.NewRequestWithContext(client.ctx, http.MethodGet, url, nil)
 	if err != nil {
-		return fmt.Errorf("error during http.NewRequestWithContext: %w", err)
+		return fmt.Errorf("http.NewRequestWithContext: %w", err)
 	}
+
 	resp, err := http.DefaultClient.Do(req)
 	if err != nil {
-		return fmt.Errorf("error during HTTP call: %w", err)
+		return fmt.Errorf("http.DefaultClient.Do: %w", err)
 	}
 	defer resp.Body.Close()
 
-	repoFile, err := os.OpenFile(repoFilename, os.O_CREATE|os.O_WRONLY|os.O_TRUNC, 0o644)
+	// Create a temp file. This automaticlly appends a random number to the name.
+	repoFile, err := ioutil.TempFile("", repoFilename)
```

Comment:
You'll hate me for this , but this will end up creating multiple new files :) Either:

1. Delete old files at the beginning of `InitRepo` call
2. Keep a single instance of the TempFile and use it across all `InitRepo` calls
3. If absolutely necessary go back to the old `ReleaseRepo` API call and we can figure how to clean up the API at a later point.

Reply:
my bad. You're right. I'll take option 1, since 2 requires closing the file in a `ReleaseRepo()` function.

Reply:
see https://github.com/ossf/scorecard/pull/599. (Sorry, the PR got auto-merged)